bears/general/LicenseCheckBear.py

```diff
@@ -0,0 +1,48 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.DistributionRequirement import (
+    DistributionRequirement)
+
+
+@linter(executable='licensecheck',
+        output_format='regex',
+        output_regex=r'(?P<message>\S+.*)')
+class LicenseCheckBear:
+    """
+    Attempts to determine the license that applies to each file passed to
+    it, by searching the start of the file for text belonging to various
+    licenses.
+    """
+    LANGUAGES = {'ALL'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='devscripts')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'License', 'Copyright Notice'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         licensecheck_lines: int=60,
+                         licensecheck_ignore: str='',
+                         licensecheck_check: str='',
+                         licensecheck_copyright: bool=False):
+        """
+        :param licensecheck_lines:
+            Specify how many lines of the file header should be parsed
+            for license information.
+        :param licensecheck_ignore:
+            Specify that files/directories matching the regular expression
+            should be ignored when checking files.
+        :param licensecheck_check:
+            Specify a pattern indicating which files should be checked.
+        :param licensecheck_copyright:
+            Display the file's copyright
```

Comment:
We were actually planning to write a bear that would check for copyright notices in source files as mentioned [here](https://github.com/savoirfairelinux/flake8-copyright). However we have wrapped this functionality in a tool called as `licensecheck` which determines the license that applies to each file and checks for copyright notices too (displays it if present).